libimage/pull.go

```diff
@@ -74,6 +75,17 @@ func (r *Runtime) Pull(ctx context.Context, name string, pullPolicy config.PullP
 			return nil, err
 		}
 		ref = dockerRef
+		possiblyUnqualifiedName = name
+	} else if ref.Transport().Name() == registryTransport.Transport.Name() {
+		// Normalize the input if we're referring to the docker
+		// transport directly. That makes sure that a `docker://fedora`
+		// will resolve directly to `docker.io/library/fedora:latest`
+		// and not be subject to short-name resolution.
+		named, err := reference.ParseNormalizedNamed(strings.TrimPrefix(name, "docker://"))
```

Comment:
This can be cheaper with `ref.DockerReference()`.

Reply:
Thanks! I thought that would only return a reference, but it really does return a normalized named one.

Reply:
Basically c/image, expect for shortnames, assumes that `docker.Reference` always comes from `ParseNormalizedNamed` (i.e. that `reference.Domain(ref)` is never empty).

It is probably possible to pass a short-name one to the API via `docker.NewReference`, but the `alltransports` parser will not do that.